src/GraphQL/Language/AST/Fragments.cs

```diff
@@ -19,6 +19,11 @@ public void Add(FragmentDefinition fragment)
             _fragments.Add(fragment ?? throw new ArgumentNullException(nameof(fragment)));
         }
 
+        /// <summary>
+        /// Returns the number of fragment definition nodes in the list.
+        /// </summary>
+        public int Count => _fragments.Count;
```

Comment:
```suggestion
        public int Count => _fragments?.Count ?? 0;
```
\+
`private readonly List<FragmentDefinition> _fragments = new List<FragmentDefinition>(); -> private List<FragmentDefinition> _fragments;`
\+
`_fragments.Add(fragment ?? throw new ArgumentNullException(nameof(fragment))); -> (_fragments ??= new List<FragmentDefinition>()).Add(fragment ?? throw new ArgumentNullException(nameof(fragment)));`
\+
`public IEnumerator<FragmentDefinition> GetEnumerator() => (_fragments ?? Enumerable.Empty<FragmentDefinition>()).GetEnumerator();`
+
```
  public FragmentDefinition FindDefinition(string name)
        {
            // DO NOT USE LINQ ON HOT PATH
            if (_fragments != null)
            {
                foreach (var f in _fragments)
                {
                    if (f.Name == name)
                        return f;
                }
            }

            return null;
        }
```